cypress/e2e/gui/e2e.cy.js

```diff
@@ -0,0 +1,113 @@
+describe('EngageSphere e2e Tests', () => {
+  beforeEach(() => {
+    Cypress.on('window:before:load', window => {
+      window.document.cookie = 'cookieConsent=accepted'
+    })
+
+    cy.visit('/');
+  });
+  
+  it('Mantém os filtros ao voltar da visualização de detalhes do cliente', () => {
+    cy.filterClient();
+    cy.get('tbody > :nth-child(1) > :nth-child(6)').click();
+    cy.contains('button', 'Back').click();
+    cy.validateFilter();
+  });
+
+  it('Retorna à lista de clientes ao clicar no botão "Voltar"', () => {
+    cy.filterClient();
+    cy.get('tbody > :nth-child(1) > :nth-child(6)').click();
+    cy.contains('button', 'Back').click();
+    cy.get('[data-testid="table"]').should('contain', 'Below is our customer list.')
```

Comment:
Que tal algo mais simples?

```suggestion
    cy.contains('[data-testid="table"]', 'Below is our customer list.')
      .should('be.visible')
```

Além dessa solução ser mais simples, perceba também a verificação pela visibilidade do elemento.

Recomendo rever a [_live_ sobre _assertions_ no YouTube](https://youtube.com/live/NBCILY2qsxQ?feature=share).